core/runtime.rs

```diff
@@ -2433,16 +2433,21 @@ impl JsRuntime {
       let mut args: SmallVec<[v8::Local<v8::Value>; 32]> =
         SmallVec::with_capacity(32);
 
-      while let Poll::Ready(Some(item)) =
-        context_state.pending_ops.poll_next_unpin(cx)
-      {
-        let (promise_id, op_id, mut resp) = item;
+      loop {
+        let mut join_fut = context_state.pending_ops.join_next();
+        // SAFETY: we are sure that the future won't move while it's being polled
+        let mut pinned_join_fut = unsafe { Pin::new_unchecked(&mut join_fut) };
+        let Poll::Ready(Some(item)) = pinned_join_fut.poll_unpin(cx) else {
+          break;
+        };
```

Comment:
This doesn't fix the crash, but doesn't require `unsafe`:

```suggestion
        let item = {
          let next = std::pin::pin!(context_state.pending_ops.join_next());
          let Poll::Ready(Some(item)) = next.poll(cx) else {
            break;
          };
          item
        };
```